src/plugins/upf/upf_pfcp_api.c

```diff
@@ -855,6 +856,23 @@ static int handle_create_pdr(upf_session_t *sess, pfcp_create_pdr_t *create_pdr,
 	      break;
 	    }
 	}
+
+	if (ISSET_BIT(pdr->pdi.grp.fields, PDI_APPLICATION_ID))
+		{
+			create->pdi.fields |= F_PDI_APPLICATION_ID;
+
+			pfcp_application_id_t *app_id = NULL;
+			pfcp_application_id_t *app_id_vector = NULL;
+	
+			vec_foreach(app_id, pdr->pdi.application_id)
```

Comment:
The PDI Application Id field is not an array, there can only be one App Id in a PDI (see Table 7.5.2.2-2 in 3GPP TS 29.244)